proposals/NNNN-lifetime-dependency.md

```diff
@@ -0,0 +1,481 @@
+let a: Array<Int>
+let ref1 = a.bufferReference() // ref1 cannot outlive a
+let ref2 = ref1.drop(4) // ref2 also cannot outlive a
+```
+
+After `ref1.drop(4)`, the lifetime of `ref2` does not depend on `ref1`.
+Rather, `ref2` has **inherited** or **copied** `ref1`’s dependency on the lifetime of `a`.
+
+#### Allowed Lifetime Dependencies
+
+The previous sections described **scoped lifetime dependencies** and **copied lifetime dependencies**
+and showed how each type occurs naturally in different use cases.
+
+Now let's look at the full range of possibilities for explicit constraints.
+The syntax is somewhat different for functions and methods, though the basic rules are essentially the same.
+
+**Functions:** A simple function with an explicit lifetime dependency annotation generally takes this form:
+
+```swift
+func f(arg: <parameter-convention> ArgType) -> @dependsOn(arg) ResultType
+```
+
+Where
+
+*  *`parameter-convention`* is one of the ownership specifiers **`borrowing`**, **`consuming`**, or **`inout`**, (this may be implied by Swift’s default parameter ownership rules),
+* `ResultType` must be `~Escapable`.
+
+If the `ArgType` is `Escapable`, the return value will have a new scoped dependency on the argument.
+(This is the only possibility, as an `Escapable` value cannot have an existing lifetime dependency,
+so we cannot copy it.)
+A scoped dependency ensures the argument will not be destroyed while the result is alive.
+Also, access to the argument will be restricted for the lifetime of the result following Swift's usual exclusivity rules:
+
+* A `borrowing` parameter-convention extends borrowing access, prohibiting mutations of the argument.
+* An `inout` parameter-convention extends mutating access, prohibiting any access to the argument.
+* A `consuming` parameter-convention is illegal, since that ends the lifetime of the argument immediately.
+
+If the `ArgType` is `~Escapable`, then it can have a pre-existing lifetime dependency.
+In this case, the semantics of `@dependsOn()` are slightly different:
+* A `consuming` parameter-convention will copy the lifetime dependency from the argument to the result
+* A `borrowing` or `inout` parameter-convention can either copy the lifetime dependency or create a new scoped lifetime dependency.
+  In this case, for reasons explained earlier, we default to copying the lifetime dependency.
+  If a scoped lifetime dependency is needed, it can be explicitly requested by adding the `scoped` keyword:
+  
+```swift
+func f(arg: borrowing ArgType) -> @dependsOn(scoped arg) ResultType
+```
+
+**Methods:** Similar rules apply to `self` lifetime dependencies on methods.
+Given a method of this form:
+
+```swift
+<mutation-modifier> func method(... args ...) -> @dependsOn(self) ResultType
+```
+
+The behavior depends as above on the mutation-modifier and whether the defining type is `Escapable` or `~Escapable`.
+
+**Initializers:** An initializer can define a lifetime dependency on one of its arguments.
```

Comment:
on one of its arguments -> on any of its arguments

Reply:
Changed.